packages/synapse-react-client/src/components/SynapseTable/SynapseTable.integration.test.tsx

```diff
@@ -0,0 +1,611 @@
+            */
+    renderTable()
+    await waitFor(() => {
+      expect(queryContext).toBeDefined()
+    })
+
+    // simulate having clicked the sort button on the first column, projectName
+    const sortedColumn = 'projectName'
+
+    const sortButton = (
+      await screen.findAllByRole('button', { name: 'sort' })
+    )[0]
+    await userEvent.click(sortButton)
+
+    // below we match only the part of the object that we expect to have changed
+    await waitFor(() => {
+      expect(queryContext!.currentQueryRequest).toEqual(
+        expect.objectContaining({
+          query: expect.objectContaining({
+            sort: [
+              {
+                column: sortedColumn,
+                direction: 'ASC',
+              },
+            ],
+          }),
+        }),
+      )
+    })
+
+    // simulate second sort click
+    await userEvent.click(sortButton)
+    // below we match only the part of the object that we expect to have changed
+    await waitFor(() => {
+      expect(queryContext!.currentQueryRequest).toEqual(
+        expect.objectContaining({
+          query: expect.objectContaining({
+            sort: [
+              {
+                column: sortedColumn,
+                direction: 'DESC',
+              },
+            ],
+          }),
+        }),
+      )
+    })
+
+    // simulate third sort click -- the sort will be removed
+    await userEvent.click(sortButton)
+    // below we match only the part of the object that we expect to have changed
+    await waitFor(() => {
+      expect(queryContext!.currentQueryRequest.query.sort).not.toBeDefined()
+    })
+  })
+
+  it('Hides download columns when rows of an entity-containing view have no IDs', () => {
+    // e.g. when the view has a GROUP BY or DISTINCT clause, the rows no longer represent individual entities, so they can't be downloaded
+    // this is indicated by the rows of the result query not having rowIds, rather than the rowId matching the synID of the corresponding entity
+    const queryResultBundleWithoutRowIds = cloneDeep(queryResultBundle)
+    queryResultBundleWithoutRowIds.queryResult!.queryResults.rows =
+      queryResultBundleWithoutRowIds.queryResult!.queryResults.rows.map(
+        row => ({
+          ...row,
+          rowId: undefined,
+        }),
+      )
+
+    // Return a file view entity, so the download column would be shown if not for the missing row IDs
+    server.use(
+      rest.get(
+        `${getEndpoint(
+          BackendDestinationEnum.REPO_ENDPOINT,
+        )}/repo/v1/entity/${synapseTableEntityId}`,
+        (req, res, ctx) => {
+          return res(ctx.status(200), ctx.json(mockFileViewEntity))
+        },
+      ),
+      ...getHandlersForTableQuery(queryResultBundleWithoutRowIds),
+    )
+
+    renderTable({ showDirectDownloadColumn: true })
+    mockAllIsIntersecting(true)
+
+    expect(
+      screen.queryByTestId('AddToDownloadListCell'),
+    ).not.toBeInTheDocument()
+  })
```

Comment:
This test might not make much sense anymore since we can add an arbitrary column